python/src/Dict2Map.hh

```diff
@@ -0,0 +1,76 @@
+#include "boost/python.hpp"
+namespace bpy = boost::python;
+
+/// This template encapsulates the conversion machinery.
+template<typename key_t, typename val_t>
+struct Dict2Map {
+
+    /// The type of the map we convert the Python dict into
+    typedef std::map<key_t, val_t> map_t;
+
+    /// constructor
+    /// registers the converter with the Boost.Python runtime
+    Dict2Map() {
+        bpy::converter::registry::push_back(
+            &convertible,
+            &construct,
+            bpy::type_id<map_t>()
+#ifdef BOOST_PYTHON_SUPPORTS_PY_SIGNATURES
```

Comment:
Was this code copied from somewhere?  What is BOOST_PYTHON_SUPPORTS_PY_SIGNATURES?